### PR TITLE
[CWS] Fix proc container ID parsing (which was too loose)

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

Tl;dr!: This PR fixes the container ID parsing from procfs, avoiding wrong containerID's to be returned.

Long version:

While debug another issue I got some error logs like:
```
2024-01-12 16:41:36 CET | SYS-PROBE | ERROR | (pkg/security/security_profile/dump/manager.go:457 in HandleCGroupTracingEvent) | couldn't start tracing [container_id:e746ad53-f64f-4ae0-b541-928a]: couldn't insert new dump: couldn't push activity dump container ID e746ad53-f64f-4ae0-b541-928a: marshal key: string doesn't marshal to 64 bytes
2024-01-12 16:41:37 CET | SYS-PROBE | ERROR | (pkg/security/security_profile/dump/manager.go:457 in HandleCGroupTracingEvent) | couldn't start tracing [container_id:9029dab8-8dfa-43ad-9ba8-20c2]: couldn't insert new dump: couldn't push activity dump container ID 9029dab8-8dfa-43ad-9ba8-20c2: marshal key: string doesn't marshal to 64 bytes
2024-01-12 16:41:47 CET | SYS-PROBE | ERROR | (pkg/security/security_profile/dump/manager.go:457 in HandleCGroupTracingEvent) | couldn't start tracing [container_id:e5f007db-4dd4-4314-962d-7bb6]: couldn't insert new dump: couldn't push activity dump container ID e5f007db-4dd4-4314-962d-7bb6: marshal key: string doesn't marshal to 64 bytes
```

And indeed they were no such containers running, but grepping proc I found:

```
$ sudo grep --color -E "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}" /proc/*/cgroup|head -n 3
/proc/101735/cgroup:1:name=systemd:/user.slice/user-1000.slice/user@1000.service/apps.slice/apps-org.gnome.Terminal.slice/vte-spawn-e0598b5b-3cd4-41a7-91cf-a6a0d96c9cff.scope
/proc/101735/cgroup:0::/user.slice/user-1000.slice/user@1000.service/apps.slice/apps-org.gnome.Terminal.slice/vte-spawn-e0598b5b-3cd4-41a7-91cf-a6a0d96c9cff.scope
/proc/1042769/cgroup:1:name=systemd:/user.slice/user-1000.slice/user@1000.service/apps.slice/apps-org.gnome.Terminal.slice/vte-spawn-e746ad53-f64f-4ae0-b541-928a65ac1e86.scope
```

Some cgroups entries can have a valid entry within their full path, leading to try to learn them.
This PR enforce the pattern matching to avoid that.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Comment by YoannGh** [comment]
The new regex probably breaks the retrieval of docker container IDs because when calling the `FindContainerID(s string)` function, the string parameter will be something like `/docker/18f0c392a6b43af80762fd3954bb91e4eaf5025a54b7edabd55bef8c2abeb959`, which doesn't match the new regex
